grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -361,6 +363,10 @@ class ProductService {
                 if (params.unitOfMeasure) ilike("unitOfMeasure", "%" + params.unitOfMeasure + "%")
                 if (params.createdById) eq("createdBy.id", params.createdById)
                 if (params.updatedById) eq("updatedBy.id", params.updatedById)
+                and {
+                    if (params.createdAfter) ge("dateCreated", dateCreatedAfter)
+                    if (params.createdBefore) le("dateCreated", dateCreatedBefore)
+                }
```

Comment:
What happens if both are null? Should we have an if (params.createdAfter or params.createdBefore) around this?

Reply:
If both are `null` from what I can tell it doesn't effect the query nor does it throw any errors, but if you have concerns that maybe this can resurface as a problem on grails 3 branch then I guess wrapping this inside of an if statement could be much safer.